### PR TITLE
Fix recognition of articles and getting PDFs in Nature

### PR DESCRIPTION
**Pull request by adam3smith**
(this was reported on the Zotero forums, but can't find it (?)

### CODE REVIEW

File: Nature Publishing Group.js [hunk 1]
```diff
@@ -651,14 +658,20 @@ function scrape(doc, url) {
 			item.publicationTitle = 'Nature';// old articles mess this up
 		}
 		delete item.journalAbbreviation;
-		
-		if (!item.attachments) {
+		var hasPDF = false;
```
Comment (adam3smith) [inline Nature Publishing Group.js]: this is potentially the most problematic bit

**Comment by zuphilip** [comment]
I guess that it fixes #1763 and maybe this is the report you remembered?

File: Nature Publishing Group.js [hunk 2]
```diff
@@ -269,9 +269,12 @@ function getKeywords(doc) {
 }
 
 //get PDF url
-function getPdfUrl(url) {
+function getPdfUrl(doc, url) {
 	var m = url.match(/(^[^#?]+\/)(?:full|abs)(\/[^#?]+?\.)[a-zA-Z]+(?=$|\?|#)/);
 	if (m && m.length) return m[1] + 'pdf' + m[2] + 'pdf';
+	else if (attr(doc, 'a[data-track-action="download pdf"]', 'href')) {
+		return attr(doc, 'a[data-track-action="download pdf"]', 'href');
```
Comment (zuphilip) [inline Nature Publishing Group.js]: Is the check before here needed? `attr` should return `null` if the element is not found, which should be okay here.

File: Nature Publishing Group.js [hunk 3]
```diff
@@ -354,7 +361,7 @@ function scrapeRIS(doc, url, next) {
 	if (!risURL) risURL = doc.evaluate('//li[@class="download-citation"]/a', doc, null, XPathResult.ANY_TYPE, null).iterateNext();
 	if (!risURL) risURL = doc.evaluate('//a[normalize-space(text())="Export citation" and not(@href="#")]', doc, null, XPathResult.ANY_TYPE, null).iterateNext();
 	if (!risURL) risURL = ZU.xpath(doc, '//ul[@data-component="article-info-list"]//a[@data-track-source="citation-download"]')[0];
-
+	if (!risURL) risURL = doc.querySelectorAll('a[data-track-action="download article citation"]')[0];
```
Comment (zuphilip) [inline Nature Publishing Group.js]: For the first element you can also use `.querySelector` cf. https://developer.mozilla.org/de/docs/Web/API/Document/querySelector

**Comment by adam3smith** [comment]
OK, all done

**Comment by zuphilip** [comment]
Thank you very much!

**Pull request merged**